### PR TITLE
fixed sign implicit conversion warnings in new enum_net code

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/enum_net.cpp [hunk 1]
```diff
@@ -178,10 +178,9 @@ namespace {
 	{
 		if (family == AF_INET)
 		{
-			using bytes_t = boost::asio::ip::address_v4::bytes_type;
-			bytes_t b;
-			std::memset(&b[0], 0xff, b.size());
-			for (int i = int(b.size()) - 1; i > 0; --i)
+			address_v4::bytes_type b;
+			b.fill(0xff);
+			for (std::size_t i = b.size() - 1; i > 0; --i)
```
Comment (aldenml) [inline src/enum_net.cpp]: hi @ssiloti, I want to confirm that `i > 0` (with `>`) is the correct condition, thanks

Comment (ssiloti) [reply src/enum_net.cpp]: No, it's not. The loop needs to run when i = 0 to set the least significant byte's value correctly if `bits` is less than 8 bits smaller than the size of the address.

Comment (aldenml) [reply src/enum_net.cpp]: I will change that then

File: src/enum_net.cpp [hunk 2]
```diff
@@ -174,39 +174,37 @@ namespace {
...
 	{
 		if (family == AF_INET)
 		{
-			using bytes_t = boost::asio::ip::address_v4::bytes_type;
-			bytes_t b;
-			std::memset(&b[0], 0xff, b.size());
-			for (int i = int(b.size()) - 1; i > 0; --i)
+			address_v4::bytes_type b;
+			b.fill(0xff);
+			for (int i = int(b.size()) - 1; i >= 0; --i)
```
Comment (aldenml) [inline src/enum_net.cpp]: @ssiloti do it looks good now? (with `i >= 0`)

Comment (ssiloti) [reply src/enum_net.cpp]: lgtm

**Pull request merged**